### PR TITLE
Prevent Button Spamming to Avoid Code Crash in Memento update learning status

### PR DESCRIPTION
**Pull request by TuanDangNguyen2003**
### PR Description
This PR addresses an issue where repeated, rapid button presses could lead to a code crash. By implementing a loading mechanism, the button's functionality is now restricted to prevent unintended rapid-fire events that previously caused instability. 

**Changes include:**
- Added a loading state for the button to limit the frequency of click
- Minor code refactor for improved readability and maintainability

This fix should enhance the app's stability and prevent crashes from excessive user interaction with the button. Please review the changes and test the button behaviour to ensure smooth functionality.

Please see how the buttons behave now right below:

https://github.com/user-attachments/assets/0b31e264-09bd-46c2-9095-fa3d90f4a541

### CODE REVIEW

File: edweiss-app/components/memento/CardScreenComponent.tsx [hunk 1]
```diff
@@ -191,9 +200,15 @@ const CardScreenComponent: ReactComponent<{ deckId: string, cardIndex: number; i
                 </TView>
                 <TView style={styles.lastButtonHalf}>
                     <FancyButton
+                        loading={isLoading}
                         testID='gotItButton'
                         backgroundColor='green'
-                        onPress={() => updateLearningStatusCard(deckId, cardIndex, updateCard, "Got it", card)}
+                        onPress={async () => {
+                            setIsLoading(true);
```
Comment (FerdinandGomerChiu) [inline edweiss-app/components/memento/CardScreenComponent.tsx:207]: You may want to create a function for this, just like you did for updateLearningStatusCard.

**Comment by TuanDangNguyen2003** [comment]
My previous PR didn't pass the TS check so I am using this branch to patch it up. I added an argument for a component and forgot to update other components that call it. It works now.

File: edweiss-app/components/memento/CardScreenComponent.tsx (hunk 1 above)
Comment (TuanDangNguyen2003) [reply edweiss-app/components/memento/CardScreenComponent.tsx:207]: I can do that so it is more readable but at the same time I think that this function is quite short and it is clear what it is doing so in my opinion, we don't need to create an extra function for this. But if there is a more complex, bigger function, it is absolutely a must to create a function outside to increase the readability of the code.

**Review by FerdinandGomerChiu — APPROVED** [review]
This PR is on the briefer side (as they should you may say). It could have been 'fused' with your previous PR. Great work still!!

**Pull request merged**